README.md

```diff
@@ -13,31 +13,33 @@ This allows you to pass around an interface containing only the code you need
 which can greatly reduce dependencies and bundle size.
 
 ```js
-import { create } from 'multiformats'
-import sha2 from 'multiformats/hashes/sha2'
+import * as CID from 'multiformats/cid'
+import { sha256 } from 'multiformats/hashes/sha2'
 import dagcbor from '@ipld/dag-cbor'
-const { multihash, multicodec, CID } = create()
-multihash.add(sha2)
-multicodec.add(dagcbor)
+import { base32 } from 'multiformats/bases/base32'
+import { base58btc } from 'multiformats/bases/base58'
 
-const buffer = multicodec.encode({ hello, 'world' }, 'dag-cbor')
-const hash = await multihash.hash(buffer, 'sha2-256')
+const bytes = dagcbor.encode({ hello: 'world' })
+
+const hash = await sha256.digest(bytes)
 // raw codec is the only codec that is there by default
-const cid = new CID(1, 'raw', hash)
+const cid = CID.create(1, dagcbor.code, hash, {
+  base: base32,
+  base58btc
+})
```

Comment:
what does this mean exactly?

i think i’d prefer this to be `{ base32, base58btc }`

we actually need the default behavior of `toString()` without a requested base encoding to be stable, so we shouldn’t have a way to set a different default base encoding here. so simply passing in the implementations should be sufficient.

Reply:
> what does this mean exactly?

Comments in the [./cid/interface.ts](https://github.com/multiformats/js-multiformats/pull/35/files#diff-921406b1f43c829e26bba1d637bf9a98) attempt to clarify that. Inlining here for convenience:

```ts
export interface Config {
  /**
   * Multibase codec used by CID to encode / decode to and out of
   * string representation.
   */
  base: MultibaseCodec<any>
  /**
   * CIDv0 requires base58btc encoding decoding so CID must be
   * provided means to perform that task.
   */
  base58btc: MultibaseCodec<'z'>
} 
```

> i think i’d prefer this to be { base32, base58btc }

Well `base` supposed to represent base codec for whatever encoding you choose to use for the CID been created. If you make it `base32` that would mean CIDs could only be in `base32` encoding. In fact if anything `base58btc` is kind of outlier here, which only there to support `toV0` and I kind of wish passing it was unnecessary.

> we actually need the default behavior of `toString()` without a requested base encoding to be stable, so we shouldn’t have a way to set a different default base encoding here. so simply passing in the implementations should be sufficient.

Are you saying user should not be able to create CID with a different base encoding ? Or simply that decision about encoding should be deferred until `toString()` is called and it should default to `base32` if no encodnig is passed ?

If later, I understand your argument. However this config is also used for parsing CID in string representations so that when you do following:

```js
const c1 = cid.parse('mAXASIDHD1XCA2EY6PGOykj31odQK16c+rloUr1hCE+X1BKwz', {
  base: base64,
  base58btc: base58btc
})
c1.toString() // mAXASIDHD1XCA2EY6PGOykj31odQK16c+rloUr1hCE+X1BKwz
```
Would you expect `c1.toString()` to print `bafybeibrypkxbagyiy5dyy5ssi67lioubll2opvolikk6wcccps7kbfmgm` instead ?

If you expect to print in `base32` then `{ base32, base58btc }` as CID config makes sense. If you expect it to print in base64 than I'd say current config makes more sense.